### ISSUE TITLE
First breakpoint set not working

### ISSUE THREAD

**Issue by CodeFrite** [CodeFrite/gameboy-go-next-js-viewer#1]
The first breakpoint set by double clicking on a memory cell has no effect on the Gameboy.Run() execution.

**Comment by CodeFrite** [comment]
The problem comes from the /gameboy/index.tsx handleToggleBreakPointEvent method which checks if the breakpoint is present in breakpoints var of type MemoryBreakPoint, adds it / removes it, and send the adequate message through socket connection to gameboy-go server:

```tsx
const handleToggleBreakPointEvent = (event: CustomEvent<ToggleBreakPointEventDetail>) => {
    console.log("gameboy> received custom event:", event);
    // toggle the breakpoint
    const memoryName = event.detail.memoryName;

    // check if there is an entry for that memory
    const memoryBreakPointIdx = breakPointsRef.current.findIndex(
      (mbp) => mbp.memoryName === event.detail.memoryName
    );
    if (memoryBreakPointIdx !== -1) {
      console.log("entries for", event.detail.memoryName, "found");
      // look for the breakpoint
      const breakPointIdx = breakPointsRef.current[memoryBreakPointIdx].addresses.findIndex(
        (value) => value.get() === event.detail.address.get()
      );
      // if it has been found, remove it from the breakpoints array and notify the server
      if (breakPointIdx !== -1) {
        setBreakPoints(
          breakPointsRef.current.map((mbp: MemoryBreakPoint) =>
            mbp.memoryName === event.detail.memoryName
              ? {
                  memoryName: mbp.memoryName,
                  addresses: mbp.addresses.filter(
                    (addr) => addr.get() !== event.detail.address.get()
                  ),
                }
              : mbp
          )
        );
        // construct the request payload, stringify it and send it to the server
        const payload = {
          type: 21,
          data: event.detail.address.get(),
        };
        ws.current!.send(JSON.stringify(payload));
        console.log("sending remove breakpoint request:", JSON.stringify(payload));

        // ... if the breakpoint hasn't been found, add it to the breakpoints array and notify the server
      } else {
        setBreakPoints(
          breakPointsRef.current.map((mbp: MemoryBreakPoint) =>
            mbp.memoryName === event.detail.memoryName
              ? {
                  memoryName: mbp.memoryName,
                  addresses: [...mbp.addresses, event.detail.address],
                }
              : mbp
          )
        );
        // construct the request payload, stringify it and send it to the server
        const payload = {
          type: 20,
          data: event.detail.address.get(),
        };
        ws.current!.send(JSON.stringify(payload));
        console.log("sending add breakpoint request:", JSON.stringify(payload));
      }
      // there are no entries for that memory, create a new entry and add the breakpoint
    } else {
      // add the breakpoint to the breakpoints array
      setBreakPoints([
        ...breakPointsRef.current,
        { memoryName: event.detail.memoryName, addresses: [event.detail.address] },
      ]);
+      // send the breakpoint to the server
+      // construct the request payload, stringify it and send it to the server
+      const payload = {
+        type: 20,
+        data: event.detail.address.get(),
+      };
+      ws.current!.send(JSON.stringify(payload));
+      console.log("sending add breakpoint request:", JSON.stringify(payload));
    }
  };
```

**Comment by CodeFrite** [comment]
fixed in commit a8331748d322b7e92f80f28e3263c64b54d425e3
